src/manifolds/Segre.jl

```diff
@@ -0,0 +1,465 @@
+@doc raw"""
+    Segre{𝔽,V} <: AbstractManifold{𝔽}
+
+The Segre manifold
```

Comment:
I have a final question on the name:
I am fine with the the `Segre` – but there is one place, the menu entry, where you write “Segre-Veronese” (as in the paper title) and otherwise you always write Segre in the text. Should we maybe use a unified form in text?
The Type can stay as is for sure.

Reply:
It should be Segre everywhere. The Segre-Veronese is when you are symmetric in some indices. It's not that hard to describe geometrically in terms of a larger Segre manifold, it's just a totally geodesic submanifold, but I didn't implement any of this.

It's on my todo list to create the Segre-Veronese as well, but I have to think through how I want to do the `connected_by_geodesic` since there is no nice closed-form expression like for the Segre.

Reply:
It should be Segre everywhere. The Segre-Veronese is when you are symmetric in some indices. It's not that hard to describe geometrically in terms of a larger Segre manifold, it's just a totally geodesic submanifold, but I didn't implement any of this.

It's on my todo list to create the Segre-Veronese as well, but I have to think through how I want to do the `connected_by_geodesic` since there is no nice closed-form expression like for the Segre.

Reply:
Nice, I can change that. And now that you have gone through a manifold once (the first is the one that takes longest), the next one will be easier. I will adapt the places to Sagre then :)